File: CODING/SORTINGS/Bubble_Sort.c
// Bubble Sort

#include<stdio.h>
#include<stdlib.h>

void PrintArray(int *arr,int size)
{
	printf("\nArray :: ");

	for(int i=0;i<size;i++)
		printf(" %d,",arr[i]);
}

void Swap(int *p,int *q)
{
	int temp=*p;
	*p=*q;
	*q=temp;
}

void bubbleSort(int *arr,int n)
{
	for(int i=0;i<n;i++)
	{
		for(int j=i+1;j<n;j++)
		{
			if(arr[i]>arr[j])
				Swap(&arr[i],&arr[j]);
		}	
	}
}

int main()
{
	int *arr,size;

	printf("\nEnter size of the Array :: ");	
	scanf("%d",&size);	

	arr=(int*)malloc(sizeof(int)*size);	

	printf("\nEnter the Array :: ");	
	for(int i=0;i<size;i++)
		scanf("%d",&arr[i]);

	PrintArray(arr,size);
	bubbleSort(arr,size);

	printf("** Sorted Successfully **");

	PrintArray(arr,size);
}

File: Core JAVA/1. Introduction/Programs with Recursions/Bubble_sort.java
class Bubble_sort
{
	public static void Sort(int arr[],int n)
	{
		if(n==1) 
			return;

		for(int i=0;i<n-1;i++) 
		{
			if(arr[i]>arr[i+1]) 
			{
				int temp=arr[i];
				arr[i]=arr[i+1];
				arr[i+1]=temp;
			}
		}
		Sort(arr,n-1);
	}

	public static void main(String args[])
	{
		int arr[]={4,1,2,7,5,9,3};

		int n=arr.length;

		for(int i=0;i<n;i++)
			System.out.print("  "+arr[i]);

		Sort(arr,n);

		System.out.println(" \n The sorted array : ");

		for(int i=0;i<n;i++)
			System.out.print("  "+arr[i]);
	}
}
